test/fail_compilation/README.md

```diff
@@ -0,0 +1,36 @@
+# README For fail_compilation tests
+
+Each file with a .d extension is run and is expected to fail to compile.
+The diagnostic emitted must match the diagnostic in the test file,
+and if it does, the test passes.
+
+## Purpose
+
+The point of these files is to test that the compiler produces a correct
+diagnostic for each error message in the compiler's implementation.
+The ideal is to achieve 100% coverage of the compiler source code.
+
+A further aim is that when the compiler does fail these tests, the test case
+should be crafted to make debugging the compiler as straightforward as practical.
+
+## Guidelines
+
+* Do not import any phobos or druntime files.
```

Comment:
We do import quite a few items from druntime, don't we ?

Reply:
Only this comment left

Reply:
This explained in `test/README`. Add a link instead of copying the content.

Reply:
> We do import quite a few items from druntime, don't we ?

Implicitly, object.d and all the baggage that comes with it (and sadly std.math for ^^).  Explicitly, extern(C) bindings are OK, but anything with a body should be included in the module as a standalone test.

We are _almost_ phobos-free in test/runnable.